python/pyarrow/array.pxi

```diff
@@ -1476,12 +1476,17 @@ cdef class Array(_PandasConvertible):
 
     def _to_pandas(self, options, types_mapper=None, **kwargs):
         pandas_dtype = None
-        try:
-            pandas_dtype = self.type.to_pandas_dtype()
-        except NotImplementedError:
-            pass
 
-        # pandas ExtensionDtype that implements conversion from pyarrow
+        if types_mapper:
+            pandas_dtype = types_mapper(self.type)
+        elif issubclass(type(self.type), ExtensionType):
```

Comment:
```suggestion
        elif self.type.id == _Type_EXTENSION:
```

Reply:
Much cleaner! I'll add to ChunkedArray as well.